cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -406,6 +406,7 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
 		if err != nil {
 			klog.Errorf("AutoscalingStatusProcessor error: %v.", err)
 		}
+		a.clusterStateRegistry.UpdateSafeScaleUpMetricsForNodeGroup(currentTime)
```

Comment:
This would better fit as an `AutoscalingStatusProcessor` implementation - all the relevant params are already passed in the `Process` call above. Let's update this function:

https://github.com/kubernetes/autoscaler/blob/a3a29cf5ddfbfffd6518af9a89b644fbb31c895f/cluster-autoscaler/processors/status/autoscaling_status_processor.go#L33-L35

to return a new processor that does the metric update. The new processor would also be a better place to keep the `backoffReasonStatus` map - `clusterStateRegistry` should be responsible for tracking the cluster state, not for metrics handling.

Reply:
Modifications done and PR submitted